<?xml version="1.0"?>
<launch>
    
    <!-- Launch file arguments -->
    <arg name="location"    default="kip"/>

    <arg name="nats"        default="false"/>

    <arg name="name"        default="rsu"/>
    
    <!-- Start NATS -->
    <node if="$(arg nats)" name="nats_connector" pkg="nats_ros_connector" type="nats_connector.py" output="screen">
        <remap from="sensor/objects" to="objectposes"/>
        <remap from="sensor/markers" to="objectmarkers"/>
        <rosparam>
        host: sml-wan.asuscomm.com
        publishers:
          - sensor/objects
          - sensor/markers
        </rosparam>
    </node>

    <!-- Start map server -->
    <arg name="map" value="sml"/>
    <node if="$(eval location == 'sml')" name="map_server" pkg="map_server" type="map_server" args="$(find svea_core)/maps/$(arg map).yaml" output="screen"/>


    <node if="$(eval location == 'kip')" pkg="tf" type="static_transform_publisher" name="map_broadcaster" output="screen" 
        args="0 0 0 0 0 0 utm map 100"/>

    <node if="$(eval location == 'kip')" pkg="tf" type="static_transform_publisher" name="odom_broadcaster" output="screen" 
        args="333308.328258 6582795.090358 0 0 0 0 map odom 100"/>

    <!-- Motion Capture System -->
    <group if="$(eval location == 'sml')">
        <arg name="mocap_address" value="10.0.0.10"/>
        <arg name="mocap_name"    value="$(arg name)"/>

        <include file="$(find mocap_qualisys)/launch/qualisys.launch">
            <arg name="server_address" value="$(arg mocap_address)"/>
        </include>

        <include file="$(find svea_mocap)/launch/map_correction.launch">
            <arg name="map" value="$(arg map)"/>
        </include>

        <!-- Static Transforms-->
        <include file="$(find svea_sensors)/launch/transforms.launch"/>
        <node pkg="tf" type="static_transform_publisher" name="sensor_broadcaster" output="screen" args="0 0 0 0 0 0 rsu zed_base_link 100"/>
    </group>

    <!--
    <node if="$(eval location == 'sml')" name="transform_sensor_objects" pkg="aug_demo" type="transform_sensor_objects.py" output="screen">
        <param name="in_top" value="objectposes"/>
        <param name="out_top" value="/sensor/objects"/>
        <param name="to_frame" value="mocap"/>
    </node>
    -->
    
    <include file="$(find svea_vision)/launch/zed_main.launch">
        <arg name="enable_bbox_image"   value="true"/>
        <arg name="aruco_size"          value="0.365"/>
        <arg name="frame_id"            value="mocap" if="$(eval location == 'sml')"/>
        <arg name="frame_id"            value="utm" if="$(eval location == 'kip')"/>
    </include>

</launch>
